lib/client.js

```diff
@@ -1097,6 +1112,17 @@ function connect (client) {
   }
 
   client[kConnecting] = true
+
+  if (channels.beforeConnect.hasSubscribers) {
+    channels.beforeConnect.publish({
```

Comment:
you may need to attach the client.  i tried subscribing but I had no object to attach some context in the connected and/or connectError to properly end spans

Reply:
just to be clear, the client will need to be on all 3 connect events

Reply:
Probably more appropriate to attach he connector?

Reply:
looks like the connector is a function, if so then I would only need to subscribe to beforeConnect and properly bind the async function

Reply:
The only benefit from passing the client, is that both `connect` and `sendHeaders` will have the same context
If we will pass connector instead of client then I would not be able to send it to `sendHeaders` 

@bizob2828 what do u prefer?

Reply:
I don't need the same context between sendHeaders and connect, But I'm fine with passing the client.  I'm very new to this lib and I just built instrumentation. @ronag appears to be involved more heavily so I will defer to him

Reply:
If we pass the connector in beforeConnect I don't see any value add with adding the connector in the connected and connectError. Typically you'd be binding the function in the beforeConnect and the other two events will be handled in the instrumentation primitives of the given vendor

Reply:
> If we pass the connector in beforeConnect I don't see any value add with adding the connector in the connected and connectError. Typically you'd be binding the function in the beforeConnect and the other two events will be handled in the instrumentation primitives of the given vendor

I see the maintainers have approved. don't sweat this. it'll still work and depending on vendor methods of attaching context this will work just fine